### ISSUE TITLE
Accessibility: Product gallery modal doesn't trap keyboard focus.

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43605]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

On the single product page, when the product image gallery is opened, it does not trap keyboard focus, making it possible to tab out of and beyond the modal, which can be confusing for screen reader or keyboard-only users.

### Expected behavior

Modals and popups should have a keyboard trap so that users cannot move beyond them on the page with the tab or arrow keys unless they are first closed.

Write JS to trap keyboard focus and add [role="dialog"](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role) to the container for the gallery.

In a modal, hitting Tab should cyle you through the focusable elements in the modal and not allow you to leave it.

### Actual behavior

No keyboard trap is present.

### Steps to reproduce

1. Go to the product single.
2. Using your keyboard, trigger the modal to open. Focus will shift to the close button for the modal.
3. Hit the Tab key or Shift+Tab key multiple times and observe that you're able to reach other elements on the page that are hidden by the modal.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by dinhtungdu** [comment]
Thanks for the report! I can confirm this bug and moved this to our backlog.

cc @thealexandrelara we should keep this in mind when working on the new product gallery block too.

**Comment by thealexandrelara** [comment]
Thank you for opening this issue! I just tested and can also confirm this is happening not only with the Product Image Gallery block but in the Product Gallery block as well.

**Comment by roykho** [comment]
Closed via https://github.com/woocommerce/woocommerce/pull/44439

**Comment by thealexandrelara** [comment]
@roykho  I'm going to re-open this issue because the changes on https://github.com/woocommerce/woocommerce/pull/44439 only fix the issue for the Product Gallery block. This reported issue is related to the Product Image Gallery block, which is a different block.

**Comment by Manussakis** [comment]
I have started working on this issue.
